src/formatDate.js

```diff
@@ -50,7 +51,66 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const OLD_SEPARATOR = fromFormat[3];
+  const NEW_SEPARATOR = toFormat[3];
+
+  const dateParts = date.split(OLD_SEPARATOR);
+  const fromFormatParts = fromFormat.slice(0, 3);
+  const toFormatParts = toFormat.slice(0, 3);
+
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormatParts.length; i++) {
+    switch (fromFormatParts[i]) {
+      case 'YYYY':
+      case 'YY':
+        year = dateParts[i];
+        break;
+      case 'MM':
+        month = dateParts[i];
+        break;
+      case 'DD':
+        day = dateParts[i];
+        break;
+    }
+  }
+
+  const newDateFormatArr = [];
+
+  for (let i = 0; i < toFormatParts.length; i++) {
+    switch (toFormatParts[i]) {
+      case 'YYYY':
+        if (year.length === 2) {
+          const FORMATED_YEAR = year < 30 ? `20${year}` : `19${year}`;
```

Comment:
Implicit type conversion in `year < 30` prefer `Number(year) < 30` this may cause unexpected behaviors